src/Content/Text/BBCode.php

```diff
@@ -636,4 +646,1341 @@ public static function limitBodySize($body)
+			if (substr($curl_info["content_type"], 0, 6) == "image/") {
+				$text = "[img]" . $match[1] . "[/img]";
+			} else {
+				$text = "[img]" . $match[2] . "[/img]";
+
+				// if its not a picture then look if its a page that contains a picture link
+				$body = Network::fetchUrl($match[1]);
+
+				$doc = new DOMDocument();
+				@$doc->loadHTML($body);
+				$xpath = new DomXPath($doc);
+				$list = $xpath->query("//meta[@name]");
+				foreach ($list as $node) {
+					$attr = [];
+					if ($node->attributes->length) {
+						foreach ($node->attributes as $attribute) {
+							$attr[$attribute->name] = $attribute->value;
+						}
+					}
+
+					if (strtolower($attr["name"]) == "twitter:image") {
+						$text = "[img]" . $attr["content"] . "[/img]";
+					}
+				}
+			}
+			Cache::set($match[1], $text);
+		}
+
+		return $text;
+	}
+
+	public static function cleanPictureLinks($text)
+	{
+		$return = preg_replace_callback("&\[url=([^\[\]]*)\]\[img\](.*)\[\/img\]\[\/url\]&Usi", 'self::cleanPictureLinksCallback', $text);
+		return $return;
+	}
+
+	private static function textHighlightCallback($match)
+	{
+		if (in_array(strtolower($match[1]),
+				['php', 'css', 'mysql', 'sql', 'abap', 'diff', 'html', 'perl', 'ruby',
+				'vbscript', 'avrc', 'dtd', 'java', 'xml', 'cpp', 'python', 'javascript', 'js', 'sh'])
+		) {
+			return text_highlight($match[2], strtolower($match[1]));
+		}
+		return $match[0];
+	}
+
+	/**
+	 * @brief Converts a BBCode message to HTML message
+	 *
+	 * BBcode 2 HTML was written by WAY2WEB.net
+	 * extended to work with Mistpark/Friendica - Mike Macgirvin
+	 *
+	 * Simple HTML values meaning:
+	 * - 0: Friendica display
+	 * - 1: Unused
+	 * - 2: Used for Facebook, Google+, Windows Phone push, Friendica API
+	 * - 3: Used before converting to Markdown in bb2diaspora.php
+	 * - 4: Used for WordPress, Libertree (before Markdown), pump.io and tumblr
+	 * - 5: Unused
+	 * - 6: Used for Appnet
+	 * - 7: Used for dfrn, OStatus
+	 * - 8: Used for WP backlink text setting
+	 *
+	 * @param string $text
+	 * @param bool   $preserve_nl
+	 * @param bool   $try_oembed
+	 * @param int    $simple_html
+	 * @param bool   $for_plaintext
+	 * @return string
+	 */
+	public static function convert($text, $preserve_nl = false, $try_oembed = true, $simple_html = false, $for_plaintext = false)
+	{
+		$a = get_app();
+
+		/*
+		 * preg_match_callback function to replace potential Oembed tags with Oembed content
+		 *
+		 * $match[0] = [tag]$url[/tag] or [tag=$url]$title[/tag]
+		 * $match[1] = $url
+		 * $match[2] = $title or absent
+		 */
+		$try_oembed_callback = function ($match)
+		{
+			$url = $match[1];
+			$title = defaults($match, 2, null);
+
+			try {
+				$return = OEmbed::getHTML($url, $title);
+			} catch (Exception $ex) {
+				$return = $match[0];
+			}
+
+			return $return;
+		};
+
+		// Hide all [noparse] contained bbtags by spacefying them
+		// POSSIBLE BUG --> Will the 'preg' functions crash if there's an embedded image?
```

Comment:
Is there a Daxygen identifier for a *possible bug*? Just wondering if this reminder could be more prominent or if this is our usual way to mark these places.

Reply:
This is our usual way of burying potential issues without signing, which means nobody can know who wrote this after such a move.